### ISSUE TITLE
Segfault if libusb fails to initialize

### ISSUE THREAD

**Issue by eric-wieser** [pyusb/pyusb#232]
Caused by the following
```python
class _LibUSB(usb.backend.IBackend):
    @methodtrace(_logger)
    def __init__(self, lib):
        usb.backend.IBackend.__init__(self)
        self.lib = lib
        self.ctx = c_void_p()
        _check(self.lib.libusb_init(byref(self.ctx)))

    @methodtrace(_logger)
    def _finalize_object(self):
        self.lib.libusb_exit(self.ctx)
```

If `libusb_init` returns an error, then `self.ctx` is `NULL`, and `libusb_exit(NULL)` segfaults.

**Comment by tedsta** [comment]
I think I'm also running into this issue.

Relevant portion of `faulthandler` output:
```
Fatal Python error: Segmentation fault
 Current thread 0x730ff470 (most recent call first):
   File "/usr/local/lib/python3.5/dist-packages/usb/backend/libusb1.py", line 684 in __del__
   File "/usr/local/lib/python3.5/dist-packages/usb/backend/libusb1.py", line 901 in get_backend
   File "/usr/local/lib/python3.5/dist-packages/usb/core.py", line 859 in find
```

**Comment by jonasmalacofilho** [comment]
LGTM.

@eric-wieser can you submit a patch?

**Comment by jonasmalacofilho** [comment]
Fixed with #279 ("Don't call libusb_exit on a null pointer").

Thank you!
